pkg/controller/master/storagenetwork/storage_network.go

```
@@ -0,0 +1,518 @@
+			*prometheusCopy.Spec.Replicas = int32(replicas)
+		}
+		delete(prometheusCopy.Annotations, ReplicaStorageNetworkAnnotation)
+
+		if _, err := h.prometheus.Update(prometheusCopy); err != nil {
+			logrus.Warnf("prometheus update error %v", err)
+			return false
+		}
+		return false
+	}
+
+	return true
+}
+
+func (h *Handler) checkAltermanagerStatusAndStart() bool {
+	// check alertmanager cattle-monitoring-system/rancher-monitoring-alertmanager replica
+	alertmanager, err := h.alertmanagerCache.Get(CattleMonitoringSystemNamespace, RancherMonitoringAlertmanager)
+	if err != nil {
+		logrus.Warnf("alertmanager get error %v", err)
+		return false
+	}
+
+	logrus.Infof("alertmanager: %v", *alertmanager.Spec.Replicas)
+	// check started or not
+	if *alertmanager.Spec.Replicas == 0 {
+		logrus.Infof("start alertmanager")
+		alertmanagerCopy := alertmanager.DeepCopy()
+		if replicas, err := strconv.Atoi(alertmanager.Annotations[ReplicaStorageNetworkAnnotation]); err == nil {
+			*alertmanagerCopy.Spec.Replicas = int32(replicas)
+		}
+		delete(alertmanagerCopy.Annotations, ReplicaStorageNetworkAnnotation)
+
+		if _, err := h.alertmanager.Update(alertmanagerCopy); err != nil {
+			logrus.Warnf("alertmanager update error %v", err)
+			return false
+		}
+		return false
+	}
+
+	return true
+}
+
+func (h *Handler) checkGrafanaStatusAndStart() bool {
+	// check deployment cattle-monitoring-system/rancher-monitoring-grafana replica
+	grafana, err := h.deploymentCache.Get(CattleMonitoringSystemNamespace, RancherMonitoringGrafana)
+	if err != nil {
+		logrus.Warnf("grafana get error %v", err)
+		return false
+	}
+
+	logrus.Infof("Grafana: %v", *grafana.Spec.Replicas)
+	// check started or not
+	if *grafana.Spec.Replicas == 0 {
+		logrus.Infof("start grafana")
+		grafanaCopy := grafana.DeepCopy()
+		if replicas, err := strconv.Atoi(grafana.Annotations[ReplicaStorageNetworkAnnotation]); err == nil {
```

It looks like we don't handle `err != nil` here.

---

I set default value to handle and ignore error